kong/pdk/request.lua

```diff
@@ -385,6 +386,34 @@ local function new(self)
     return s and sub(uri, 1, s - 1) or uri
   end
 
+  ---
+  -- Returns the normalized path component of the request's URL. The return
+  -- value is the same as `kong.request.get_path()` but normalized according
+  -- to RFC 3986 section 6:
+  --
+  -- * Percent-encoded values of unreserved characters are decoded (`%20`
+  --   becomes ` `).
+  -- * Percent-encoded values of reserved characters have their hexidecimal
+  --   value uppercased (`%2f` becomes `%2F`).
+  -- * Relative path elements (`/.` and `/..`) are dereferenced.
+  --
+  -- Optionally, when the `merge_slashes` param is `true`, duplicate slashes are
+  -- consolidated (`//` becomes `/`).
+  --
+  -- @function kong.request.get_normalized_path
+  -- @phases rewrite, access, header_filter, response, body_filter, log, admin_api
+  -- @tparam[opt] boolean merge_slashes consolidate duplicate slashes in the path
+  -- @treturn string the path
+  -- @usage
+  -- -- Given a request to https://example.com/t/Abc%20123%C3%B8%2f/parent/..//test/./
+  --
+  -- kong.request.get_normalized_path()     -- "/t/Abc 123ø%2F//test/"
+  -- kong.request.get_normalized_path(true) -- "/t/Abc 123ø%2F/test/"
+  function _REQUEST.get_normalized_path(merge_slashes)
+    local path = _REQUEST.get_path() or ""
+    return normalize(path, merge_slashes)
+  end
```

Comment:
A few things:
1. `_REQUEST.get_path()` does not return `false` or `nil` so `or ""` is not needed.
2. `merge_slashes` is questionable, and if we have that it should be opposite `do_not_merge_slashes` as merging them is what people in 99% cases want. That is also what we do in our router.

Reply:
Agreed, I'll make `merge_slashes` the default behavior.

Reply:
Updated. I've kept the parameter name the same (`merge_slashes`) but changed the behavior to be truth-y by default. I have a _slight_ preference for this as opposed to inverting the param to `do_not_merge_slashes`, but I don't mind switching it to `do_not_merge_slashes` with a false-y default if you feel like it's too clunky this way.

Reply:
+1 to not using negative variable names

Reply:
I agree with @bungle and would prefer the negative name here since it would stick better to the truthy/falsy behaviour expected in Lua. If one reads `merge_slashes` then most of us would expect the default to be `false`. But by no means want to turn this into a 🔥 🔥 🔥 🔥 🔥 🔥  😄